### PR TITLE
Options decorator

### PR DESCRIPTION
**Pull request by RyanYoung25**
Added a decorator to handle OPTIONS requests and removed the duplicate code from every endpoint that handles OPTIONS.

**Significant Changes**
1. There is now a decorator. It's near the top of app.py 
2. All of the endpoints that accept OPTIONS are decorated by the decorator 
3. All of the OPTIONS duplicate code has been removed

**How to Test**
1. Use the application. There is not a specific feature that this ties to just every end point so try them out in the usual way. 

** Notes **
I didn't update the version yet because it might not be accepted immediately. Will update when merging.

### CODE REVIEW

File: Backend/app.py [hunk 1]
```diff
@@ -68,6 +69,18 @@ def getDatabaseClient():
...
     return boto3.resource('dynamodb', endpoint_url='http://localhost:8000')
 
+
+def handleOptionsRequest(endpoint):
+  @functools.wraps(endpoint)
+  def optionsDecorator(*args, **kws):
+    if request.method =='OPTIONS':
+      status = 200
+      flaskResponse = ResponseCreation.createEmptyResponse(status)
+      return flaskResponse
```
Comment (jonathanmcdaniel) [inline Backend/app.py]: Could this be made `return ResponseCreation.createEmptyResponse(status=200)` to get rid of those two lines above the return?

File: Backend/app.py [hunk 2]
```diff
@@ -68,6 +69,16 @@ def getDatabaseClient():
     logger.info("Using Local Dev Database")
     return boto3.resource('dynamodb', endpoint_url='http://localhost:8000')
 
+
+def handleOptionsRequest(endpoint):
```
Comment (aaronmsegal) [inline Backend/app.py]: Please move to a utils script.

**Comment by asp78** [comment]
Don't forget to update the version number

File: Backend/app.py [hunk 3]
```diff
@@ -12,6 +12,7 @@
 from problems import algebra
 from utils import MentiiAuth
 from utils.ResponseCreation import ControllerResponse
```
Comment (jonathanmcdaniel) [inline Backend/app.py]: You have a duplicate import

File: Backend/app.py [hunk 4]
```diff
@@ -12,6 +12,7 @@
...
+from utils.decorators import *
 import utils.ResponseCreation as ResponseCreation
```
Comment (jonathanmcdaniel) [inline Backend/app.py]: This is the duplicate import

File: Backend/app.py [hunk 5]
```diff
@@ -12,6 +12,7 @@
 from problems import algebra
 from utils import MentiiAuth
 from utils.ResponseCreation import ControllerResponse
+from utils.decorators import *
```
Comment (jonathanmcdaniel) [inline Backend/app.py]: Capitalize decorators like the other module/packages/those thingys

File: Backend/app.py (hunk 4 above)
Comment (jonathanmcdaniel) [inline Backend/app.py]: `import utils.ResponseCreation as ResponseCreation`

**Pull request merged**